src/azure-cli/azure/cli/command_modules/mysql/custom.py

```diff
@@ -71,16 +70,15 @@ def flexible_server_threat_model_update(cmd, client, resource_group_name,
     if server_name is None:
         raise ValueError("Invalid server name provided.")
 
-    if advanced_threat_protection_name is None:
+    if advanced_threat_protection_name == mysql_flexibleservers.models.AdvancedThreatProtectionName.DEFAULT.name:
         raise ValueError("Invalid defender protection name provided.")
 
-    if defender_state is None:
+    if not (defender_state.lower() == mysql_flexibleservers.models.AdvancedThreatProtectionState.ENABLED.name.lower() or defender_state.lower() == mysql_flexibleservers.models.AdvancedThreatProtectionState.DISABLED.name.lower()):
```

Comment:
This is the right way to compare the values, there is a potential bug here

Reply:
Code won't handle the case sensitive case. E.g., if an user passes "default", code will not throw error but the expected behaviour is to not allow that as allowed value is "Default"